Add deleting and fix something else

AdrianKuriata opened the pull request:
Hi, i added deleting and adding languages from the GUI, changed something in the index etc. you can check it on the [Video](https://www.youtube.com/watch?v=mLrUFI32ugg&feature=youtu.be) or [ScreenShot](http://i.imgur.com/UagaOtM.png). Be happy!

Some changes for #24 

I commented annoying alert, do you think this is #11 ?

AdrianKuriata committed:
Added delete language function.

AdrianKuriata committed:
Added delete function.

AdrianKuriata committed:
Added routes for deleting language.

AdrianKuriata committed:
Couple changes in view

Added possibility to adding new language, deleting language.

AdrianKuriata committed:
Some changes

Added comment for annoying alert and added deleteLanguage function with confirmation. For addLanguage function added alert about it and reload location for fix displaying and possibility scaning files for created language(i wrote about this in my issue).

AdrianKuriata committed:
I need delete this file...

Yea, i didn't download this package with github for do changes i just add commits from github. I can't edit this file, because is to large so i upload just compiled file.

AdrianKuriata committed:
Adding new file with compiled changes

themsaid commented on resources/js/langman.js:
```diff
@@ -80,9 +80,9 @@ new Vue({
          * Add a new translation key
          */
         addNewKey(key) {
```
This method is only used to add a new translation key, adding a new language should be a separate action.

AdrianKuriata replied:
I didn't create new action for adding new language, because i just add action what was created for this by you. I just display in index.blade.php button for create new language, in this metod i commented if statement, because if you scaned files and nothing new came out this alert worked as many times as there were keys and it is annoying. Here you displaying keys into loop so this is working like that. I testing this changes and everything working perfectly.

AdrianKuriata replied:
Sorry, i give you your honor, i just misswrite alert message in wrong action, this should be in addLanguage action, not into addNewKey... I will fix it soon.

themsaid commented on resources/js/langman.js:
```diff
@@ -80,9 +80,9 @@ new Vue({
+            /*if (this.translations[this.baseLanguage][key] !== undefined) {
                 return alert('This key already exists.');
-            }
+            }*/ //Really annoying behavior...
```
Because you can't add a key multiple times, so if it already exists the alert prevents you from re-adding it.

AdrianKuriata replied:
I say something about this in previous answer. And sure, if you has hmm, 200 records to translate, this alert will be display 200 times.

Better will be do something like `var hasExistKeys = false;` and into FOR LOOP if has change `hasExistKeys = true;` and under this loop check `if(hasExistKeys){alert('This key already exists.');}`

AdrianKuriata commented:
@themsaid Ok i tested it really advanced and is problem with scan keys for new language. If you will try scan keys for new language in the `scanForKeys()` action you have forEach function which is going to addNewKey, i don't know why, but if this keys was added in other language, you can't add keys for translate new language, because you will get alert with "This key already exists" and it will end it because  this is return.

1. I fixed empty field for creating new language and adding empty file witout name, mean `.json`.
2. I added new action `addNewKeysForScan()` for `scanForKeys()` - this actions conflicted when you tried scan keys for new language(displayed information about "this key already exists" many times as many was keys). We do not need check key exists for scaning keys for new language so i needed add new action for adding keys for new language. After testing everything working perfectly.

You can test it, it should working normally, i fixed everything what could be annoying (i hope so...).

AdrianKuriata committed:
Added new improvements and fixed bugs.

AdrianKuriata committed:
Added improvements and bug fixed.

robsontenorio commented:
+1

stefro commented:
+1

themsaid commented:
Sorry I know you've put much effort into this but I don't really want to allow removing languages from GUI, also on adding new language I don't want to force refresh the screen since users might be having unsaved keys.

Thanks again for your time :)
